var/spack/repos/builtin/packages/fpocket/package.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Fpocket(Package):
+    """fpocket is a very fast open source protein pocket detection algorithm
+       based on Voronoi tessellation."""
+
+    homepage = "https://github.com/Discngine/fpocket"
+    version('develop', branch='master',
+            git='https://github.com/Discngine/fpocket.git')
+
+    depends_on("netcdf")
+
+    def setup_environment(self, spack_env, run_env):
+        if self.compiler.name == 'gcc':
+            spack_env.set('CXX', 'g++')
```

Comment:
Spack already automatically sets `CXX` and friends so you don't need to.

Reply:
It was not working for me since I was using the system gcc. So it was going for clang compiler.

Reply:
Can you share your `compilers.yaml`?

Reply:
here is were I identify the system GCC:
```compilers:
- compiler:
    environment: {}
    extra_rpaths: []
    flags:
      cflags: -march=nehalem -O3
      fflags: -march=nehalem -O3
      cxxflags: -march=nehalem -O3
      cppflags: -march=nehalem -O3
    modules: []
    operating_system: ubuntu16.04
    paths:
      cc: /usr/bin/gcc-5
      cxx: /usr/bin/g++-5
      f77: /usr/bin/gfortran-5
      fc: /usr/bin/gfortran-5
    spec: gcc@5.4.0
    target: x86_64
```

Reply:
And can you show me the error that happens when you don't set CXX?

Reply:
It wasn't exactly an error. In fpocket's makefile, it checks for CXX. If it's set to g++, it uses gcc, else it falls for clang compiler. And looks like it could't detect it from my configurations, so it always went for clang until I added that part of setting CXX.

Reply:
that's the part in makefile
```ifeq ($(CXX),g++)
  CC          = gcc
  CCQHULL           = gcc
  LINKER      = LD_LIBRARY_PATH=$(PLUGINDIR)/$(ARCH)/molfile gcc
  LINKERQHULL = gcc
else
  CCQHULL           = clang
  CC          = clang
  LINKER      = LD_LIBRARY_PATH=$(PLUGINDIR)/$(ARCH)/molfile clang
  LINKERQHULL = clang
endif
```

Reply:
Ah, I see now. I think your solution is fine, although it might be more robust to patch the Makefile so it always uses Spack's compiler wrappers, regardless of if we're using GCC, Clang, or any other compiler. It looks like no other compilers are supported unfortunately.